h/api/test/storage_test.py

```diff
@@ -9,109 +11,515 @@
+        postgres_enabled.return_value = False
+
+        request = DummyRequest()
+        document = models.elastic.Document.get_by_uri.return_value
+        type(document).document_uris = uris = mock.PropertyMock()
+        uris.return_value = [
+            mock.Mock(uri='http://foo.com/'),
+            mock.Mock(uri='http://bar.com/'),
+            mock.Mock(uri='http://example.com/', type='rel-canonical'),
+        ]
+        assert storage.expand_uri(request, "http://example.com/") == [
+            "http://example.com/"]
 
-    document = Document(document_uris=[
-        DocumentURI(uri='http://foo.com/', claimant='http://example.com'),
-        DocumentURI(uri='http://bar.com/', claimant='http://example.com'),
-        DocumentURI(uri='http://example.com/', type='rel-canonical', claimant='http://example.com'),
-    ])
-    db.Session.add(document)
-    db.Session.flush()
+    def test_expand_uri_postgres_document_uris(self, postgres_enabled):
+        request = DummyRequest(db=db.Session)
+        postgres_enabled.return_value = True
 
-    assert storage.expand_uri(request, "http://example.com/") == [
-            "http://example.com/"]
+        document = Document(document_uris=[
+            DocumentURI(uri='http://foo.com/', claimant='http://bar.com'),
+            DocumentURI(uri='http://bar.com/', claimant='http://bar.com'),
+        ])
+        db.Session.add(document)
+        db.Session.flush()
 
+        assert storage.expand_uri(request, 'http://foo.com/') == [
+            'http://foo.com/',
+            'http://bar.com/'
+        ]
 
-def test_expand_uri_elastic_document_doesnt_expand_canonical_uris(postgres_enabled, document_model):
-    postgres_enabled.return_value = False
+    def test_expand_uri_elastic_document_uris(self, postgres_enabled, models):
+        postgres_enabled.return_value = False
+        request = DummyRequest()
+        document = models.elastic.Document.get_by_uri.return_value
+        type(document).document_uris = uris = mock.PropertyMock()
+        uris.return_value = [
+            mock.Mock(uri="http://foo.com/"),
+            mock.Mock(uri="http://bar.com/"),
+        ]
+        assert storage.expand_uri(request, "http://example.com/") == [
+            "http://foo.com/",
+            "http://bar.com/",
+        ]
 
-    request = DummyRequest()
-    document = document_model.get_by_uri.return_value
-    type(document).document_uris = uris = mock.PropertyMock()
-    uris.return_value = [
-        mock.Mock(uri='http://foo.com/'),
-        mock.Mock(uri='http://bar.com/'),
-        mock.Mock(uri='http://example.com/', type='rel-canonical'),
-    ]
-    assert storage.expand_uri(request, "http://example.com/") == [
-            "http://example.com/"]
 
+@pytest.mark.usefixtures('AnnotationBeforeSaveEvent',
+                         'models',  # Don't try to talk to real Elasticsearch!
+                         'partial',
+                         'transform')
+class TestCreateAnnotation(object):
```

Comment:
I would rename this test class and the Postgres one to `TestCreateAnnotationElasticsearch/Legacy` and the Postgres one to `TestCreateAnnotation`, makes it easier to get rid of the legacy code later on.